### ISSUE TITLE
Inconsistent display of man pages (`:Man` vs `$ man`)

### ISSUE THREAD

**Issue by lacygoill** [neovim/neovim#9156]
- `nvim --version`:      NVIM v0.3.2-715-gaff64b3 Build type: RelWithDebInfo
 - Operating system:     Ubuntu 16.04.5 LTS
 - Terminal emulator:    rxvt-unicode v9.22
 - `$TERM`:              tmux-256color
 - Shell:                zsh 5.6.2-dev-0 (x86_64-pc-linux-gnu)

According to `:h ft-man-plugin`, one can use Neovim as a manpager, after executing this shell command:

    export MANPAGER="nvim -c 'set ft=man' -"

Here's how Neovim displays the manpage for `$ bash` when I use the `$ man` command:
![gif1](https://user-images.githubusercontent.com/8505073/47520697-2a84bd80-d891-11e8-94b7-4d4c6dec18c7.gif)

And here's the same page when I use the `:Man` command:
![gif2](https://user-images.githubusercontent.com/8505073/47520716-3a9c9d00-d891-11e8-9a4b-eb24c56405b8.gif)

The version printed by `:Man` is different because it supports bold text.

I think the issue comes from the fact that when `$ man` is invoked, the man filetype plugin only calls `man#init_pager()`, while when `:Man` is invoked, an autocmd calls `man#read_page()`.
The latter seems to better format a manpage.

I don't know much about the code, so there may be a better way, but I think the issue could be fixed by executing the autocmd at the end of `man#init_pager()`.
That is by adding the line `doautocmd man BufReadCmd`:

    function! man#init_pager() abort
      ...
      doautocmd man BufReadCmd
    endfunction

I tried to add the command before the end of the function, and while it still fixed the issue, it also made Neovim print the message `No matching autocommands` on the terminal, which could be seen when quitting Neovim.

    $ man bash
    No matching autocommands

I couldn't eliminate the message even with a guard to check the existence of the autocmd:

    if exists('#man#BufReadCmd')
      doautocmd man BufReadCmd
    endif

However, again, if `doautocmd man BufReadCmd` is written at the end of `man#init_pager()`, the message is not printed on the terminal.

**Comment by justinmk** [comment]
NB: Instead of `:doautocmd man BufReadCmd` you can just run `:edit` .

**Comment by lacygoill** [comment]
I don't know whether it's linked to the current issue but I noticed 2 other weird behaviors.

---

You can't open a manpage outside the first section of the manual if you've started Neovim as a manpager.

MWE:

    $ MANPAGER='nvim +Man!' man man
    :Man mount

Output:

    man.vim: command error (11) man -w mount: No manual entry for mount
    See 'man 7 undocumented' for help when manual pages are not available.

![gif](https://user-images.githubusercontent.com/8505073/47536436-2fad3100-d8c0-11e8-9cdf-a547c1749988.gif)


The issue can be simplified further:

    $ MANPAGER='nvim -Nu NONE -' man man
    :!man -w mount

Output:

    No manual entry for mount
    See 'man 7 undocumented' for help when manual pages are not available.

    shell returned 16

I don't know why, but for some reason, `$ man -w` can't find the location of a page outside the first section of the manual (unless you provide the section number manually with `-s`).
The issue is not specific to Neovim, I was able to reproduce it with Vim:

    $ MANPAGER='vim -Nu NONE -' man man
    :!man -w mount

It's not specific to `:!` either, it can be reproduced with `system()` too:

    $ MANPAGER='nvim -Nu NONE -' man man
    :echo system('man -w mount')

I don't understand why that happens, since these commands (which I thought would emulate what the shell does with `$MANPAGER`) work as expected:

    $ zcat /usr/share/man/man1/man.1.gz | nvim -
    " ✔
    :!man -w mount

---

If you start Neovim with the `$ nvim` command, then execute `:Man bash`, the name of the buffer is written in lowercase, so you see this in the statusline:

    man://bash(1)

But if you start Neovim with the `$ man bash` command, this time the name of the buffer is written in uppercase:

    man://BASH(1)

**Comment by justinmk** [comment]
> ```
> $ MANPAGER='nvim +Man!' man man
> :Man mount
> ```
>
> The issue is not specific to Neovim, I was able to reproduce it with Vim:

It works for me with `NVIM v0.3.2-752-g4d7c7f96ddcb`. Seems like your system has something unexpected by the man.vim plugin, maybe.

> But if you start Neovim with the $ man bash command, this time the name of the buffer is written in uppercase:

Nothing we can do about that. When invoked as MANPAGER we must guess the manpage and section based on the header, which is usually uppercase.  Some manpages/commands are title-cased, so we can't know the correct case. Although, it might be worth lowercasing anyways, since that's the most common.

**Comment by lacygoill** [comment]
It's possible that the issue comes from my man plugin, but I doubt it, I've copied again the latest version of the 3 files `plugin/man.vim`, `autoload/man.vim`, `lua/man.lua` from the repository, and I've also recompiled the latest Neovim version `NVIM v0.3.2-752-g4d7c7f9`.

It may come from the version of `$ man`. I'm stuck on a 3-year old system (`man --version` reports `2.7.5`). If I could update `$ man`, I could check whether this is it, but I don't know how to compile it. I'll try to find some information on Google.

**Comment by lacygoill** [comment]
The 2nd issue comes from the fact that I set the environment variable `$MANSECT` in a zsh initialization file (`~/.zshenv`):

    export MANSECT=1:n:l:8:3:2:3posix:3pm:3perl:5:4:9:6:7

When you start Neovim with the `$ nvim` command, this environment variable is not altered:

    $ nvim
    :echo $MANSECT

The last `:echo` correctly prints the value you assigned to `$MANSECT` in your shell.

But if you start Neovim with the `$ man` command, the value is truncated:

    $ man man
    :echo $MANSECT

This time, the last `:echo` doesn't print the full value of `$MANSECT`, it only prints the first token, before the first colon. All the rest is lost.

I logged Neovim's activity in a file:

    $ export MANSECT=1:n:l:8:3:2:3posix:3pm:3perl:5:4:9:6:7
    $ MANPAGER='nvim +Man! -V15/tmp/log' man man

But I couldn't find any statement which was changing the value of `$MANSECT`.

There may be a better way, but this issue could be fixed in [s:system()](https://github.com/neovim/neovim/blob/master/runtime/autoload/man.vim#L119-L149) by saving the value of `$MANSECT` and making it temporarily empty at the beginning of the function, then restoring its value at the end:

    function! s:system(cmd, ...) abort
      let mansect_save = $MANSECT
      let $MANSECT = ''
      ...
      let $MANSECT = mansect_save
    endfunction
